### PR TITLE
Segment - Fix filter URL Visited with regex and more

### LINKED ISSUE #11427: Segment with URL visited and regexp don't filter
### Mautic Version

4.4.x series

### PHP version

7.4.30

### What browsers are you seeing the problem on?

Not relevant

### What happened?

I make a segment with a filter. I use "URL visited" as field with "regexp" as filter. 
I want to know where the different contacts come from. But, where I do my different segments, with different text based on my different url, I have always the same number of contacts.
I check my regex, and I think it's ok. I just put a text. 
Moreover, I also checked, and I have contacts who don't visit all url. 

Do you any idea?

### How can we reproduce this issue?

1. Create a segment
2. Choose like filter : "email not empty" AND "URL visited regexp _one url of your site_"
3. Save and close
4. Create a other segment 
5. Choose like filter : "email not empty" AND "URL visited regexp _other url of your site_"
6. Save and close
7. Compare the number of contacts

### Relevant log output

```shell
No log
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Cr7t3K**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [Y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [Y]
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | Fixes #11427 

#### Description:

This PR fixes the problem with regex filters, like the visited url.
The filter did not take into account the regex, and display the same number of contacts no matter the regex.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Login
4. Create two segments with filter "Visited URL" and "regexp", but not same expression (depends on your current data in table page_hits)
5. Update your two segments with command
6. You will see that the filters work
7. Enjoy

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11541"></a>

PR funded by @PierreAmmeloot from [WebAnyOne](https://www.webanyone.net/)

### CODE REVIEW

**Review by PierreAmmeloot — APPROVED** [review]

File: app/bundles/LeadBundle/Segment/ContactSegmentFilterCrate.php [hunk 1]
```diff
@@ -90,6 +91,14 @@ public function isCompanyType()
         return self::COMPANY_OBJECT === $this->object;
     }
 
+    /**
+     * @return bool
+     */
+    public function isBehaviorsType()
```
Comment (escopecz) [inline app/bundles/LeadBundle/Segment/ContactSegmentFilterCrate.php]:
```suggestion
    public function isBehaviorsType(): bool
```

**Review by escopecz — CHANGES_REQUESTED** [review]
I added a small request before we merge this PR. Also, please create a duplicate PR against the 5.x branch to confirm it will work there too and all conflicts will be solved. Thanks for the functional tests! It makes me a lot more confident that it works!

**Review by escopecz — APPROVED** [review]
Thank you! 👍

**Comment by escopecz** [comment]
@Cr7t3K create please also PRs with these commits against the 5.x branch. It may need rebase or cherry-pick. See https://github.com/mautic/mautic/compare/5.x...Cr7t3K:fix-segment-filter-regex

**Comment by escopecz** [comment]
Aha, you already did. https://github.com/mautic/mautic/pull/11556 Thank you!

**Pull request merged**